src/main/java/org/cru/redegg/servlet/RedEggServletListener.java

```diff
@@ -138,4 +149,17 @@ public void setSanitizer(ParameterSanitizer sanitizer)
     {
         this.sanitizer = sanitizer;
     }
+
+    private boolean isAsyncContext(ServletRequest request) {
+        try {
+            request.getAsyncContext();
+            return true;
+        } catch (IllegalStateException e) {
+            return false;
+        }
+    }
```

Comment:
did you try out `ServletRequest.isAsyncStarted()`? It'd be nice to not use exceptions for normal flow control.